### DISCUSSION TITLE
Modify/filter  __NEXT_DATA__

### DISCUSSION THREAD

**Discussion by dipak7587** [vercel/next.js#15487] [Help] [answered]
Hi Fork,

Actually my requirement is, I want to filter initial data before adding in the script tag.

I did not find any function to modify __NEXT_DATA__ data. In that reason, my page size is too big

**Comment by jamesmosier** [comment, marked as answer]
`__NEXT_DATA__` is the data you return from `getStaticProps`, `getServerSideProps`, or `getInitialProps`. So if you want to filter the data in `__NEXT_DATA__` you'll need to filter what props you return from those methods.

**Reply by ktrip96** [reply]
But what if I want to transfer security-sensitive data through `getServerSideProps` for example?
The user will have access to that secret keys through `__NEXT_DATA_`.
